### ISSUE TITLE
Switch to an arbitrary frame in video

### ISSUE THREAD

**Issue by gena** [BinomialLLC/basis_universal#43]
I've tried to add a slider to test if quick switching between video frames would work, unfortunately, this generates fatal error very quickly and some frames look damaged as well, probably the API is used incorrectly. Would it be possible to support this kind of fast switching between frames or what do I miss in the API? I'm trying to evaluate if basis can be used to keep videos with satellite images instead of webm, to have a faster way to inspect them. Or maybe at least encode webm>basis locally to support faster inspection and editing locally.

Here is a code snippet built on top of [webgl_videotest](https://github.com/BinomialLLC/basis_universal/tree/master/webgl_videotest): https://codesandbox.io/s/static-0g1o9?fontsize=14.

**Comment by richgel999** [comment]
Yes, the current encoder only generates an I-Frame on the first frame. It would be fairly easy (probably trivial) to add support for I-Frames every X frames.

Honestly, you are the first to write about video use cases which is very exciting!

**Comment by richgel999** [comment]
We will get this feature in over the next month. We are very excited to have someone use Basis Universal for video.

**Comment by gena** [comment]
Thanks!

**Comment by dustinkerstein** [comment]
I was just curious, is it now possible to get the encoder to generate I-Frames on every frame?

**Comment by richgel999** [comment]
That's what the encoder normally does with texture arrays, unless its in texture video mode. So just pack to a texture array vs. a video.
